neurons/validators/src/services/docker_service.py

```diff
@@ -41,15 +41,33 @@ def __init__(
         self.ssh_service = ssh_service
         self.redis_service = redis_service
 
-    def generate_portMappings(self, start_external_port=40000) -> list[tuple[int, int]]:
+    def generate_portMappings(self, range_external_ports: str) -> list[tuple[int, int]]:
         internal_ports = [22, 22140, 22141, 22142, 22143]
 
         mappings = []
         used_external_ports = set()
+        
+        # Parse the range_external_ports
+        if range_external_ports:
+            if '-' in range_external_ports:
+                # Handle range like "100-1500"
+                start, end = map(int, range_external_ports.split('-'))
```

Comment:
@LegendWorrier  could you please update this that allow space between dash?

Reply:
![image](https://github.com/user-attachments/assets/189f9699-bd1c-4caa-aecc-3b7381142b57)

Reply:
@pyon12 when space between dash, it return value correctly.
![image](https://github.com/user-attachments/assets/6cae6dc4-199f-4a1a-9c21-b1fd6b6550f2)